src/test/schema.test.ts

```diff
@@ -273,6 +273,243 @@ suite('JSON Schema', () => {
 
 	});
 
+	test('Resolving $refs to local $ids', async function () {
+		const service = new SchemaService.JSONSchemaService(newMockRequestService(), workspaceContext);
+		
+		service.setSchemaContributions({
+			schemas: {
+				"https://myschemastore/main/schema1.json": {
+					id: 'https://myschemastore/main/schema1.json',
+					definitions:{
+						hello: {
+							id: '#hello',
+							type: 'string',
+							const: 'hello'
+						},
+						world: {
+							$id: '#world',
+							type: 'string',
+							const: 'world'
+						}
+					},
+					type: 'object',
+					properties: {
+						p1: {
+							$ref: '#hello'
+						},
+						p2: {
+							$ref: '#world'
+						}
+					}
+				}
+			}
+		});
+
+		return service.getResolvedSchema('https://myschemastore/main/schema1.json').then(fs => {
+			assert.deepEqual(fs?.schema.properties?.p1, {
+				type: 'string',
+				const: 'hello'
+			});
+			assert.deepEqual(fs?.schema.properties?.p2, {
+				type: 'string',
+				const: 'world'
+			});
+		});
+
+	});
+
+	test('Resolving $refs to external $ids', async function () {
+		const service = new SchemaService.JSONSchemaService(newMockRequestService(), workspaceContext);
+		service.setSchemaContributions({
+			schemas: {
+				"https://myschemastore/main/schema1.json": {
+					id: 'https://myschemastore/schema1.json',
```

Comment:
Shouldn't that be `id: 'https://myschemastore/main/schema1.json'` ?

Reply:
Yes indeed it should be the same, although it doesn't affect the test, I would keep them the same for clarity. I'll fix it by the end of this day hopefully